src/integrationTest/java/org/opensearch/security/rest/GzipTests.java

```diff
@@ -0,0 +1,115 @@
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.zip.GZIPOutputStream;
+
+import static org.junit.Assert.fail;
+import static org.opensearch.test.framework.TestSecurityConfig.AuthcDomain.AUTHC_HTTPBASIC_INTERNAL;
+import static org.opensearch.test.framework.TestSecurityConfig.Role.ALL_ACCESS;
+
+@RunWith(com.carrotsearch.randomizedtesting.RandomizedRunner.class)
+@ThreadLeakScope(ThreadLeakScope.Scope.NONE)
+public class GzipTests {
+    private static final TestSecurityConfig.User ADMIN_USER = new TestSecurityConfig.User("admin").roles(ALL_ACCESS);
+
+    @ClassRule
+    public static LocalCluster cluster = new LocalCluster.Builder().clusterManager(ClusterManager.THREE_CLUSTER_MANAGERS)
+        .authc(AUTHC_HTTPBASIC_INTERNAL)
+        .users(ADMIN_USER)
+        .anonymousAuth(false)
+        .build();
+
+    @Test
+    public void testAuthenticatedGzippedRequests() {
+        final String requestPath = "/*/_search";
+        final int parrallelism = 10;
+        final int totalNumberOfRequests = 100;
+
+        final byte[] compressedRequestBody = createCompressedRequestBody();
+        List<TestRestClient> restClients = new ArrayList<>();
+        for (int i = 0; i < 10; i++) {
+            restClients.add(cluster.getRestClient(ADMIN_USER, new BasicHeader("Content-Encoding", "gzip")));
+        }
+        Random rand = new Random();
+        final HttpPost post = new HttpPost(restClients.get(0).getHttpServerUri() + requestPath);
+        post.setEntity(new ByteArrayEntity(compressedRequestBody, ContentType.APPLICATION_JSON));
+
+        final ForkJoinPool forkJoinPool = new ForkJoinPool(parrallelism);
+
+        final List<CompletableFuture<Void>> waitingOn = IntStream.rangeClosed(1, totalNumberOfRequests)
+            .boxed()
+            .map(i -> CompletableFuture.runAsync(() -> {
+                TestRestClient.HttpResponse response = restClients.get(rand.nextInt(10)).executeRequest(post);
+                assertThat(response.getStatusCode(), equalTo(HttpStatus.SC_OK));
+                assertThat(response.getBody(), not(containsString("json_parse_exception")));
+            }, forkJoinPool))
+            .collect(Collectors.toList());
+
+        final CompletableFuture<Void> allOfThem = CompletableFuture.allOf(waitingOn.toArray(new CompletableFuture[0]));
+
+        try {
+            allOfThem.get(30, TimeUnit.SECONDS);
+        } catch (final Exception e) {
+            Throwable rootCause = e.getCause();
+            if (rootCause instanceof AssertionError) {
+                fail("Received exception: " + e.getMessage());
+            }
+        }
+    }
+
+    private byte[] createCompressedRequestBody() {
```

Comment:
Nit: Seems like we could refactor to be reusable from TestRestClient

Reply:
Sounds good, I'll push another update with this shortly. This can take a `String body` in as a param.

Reply:
I opted not to refactor ResourceFocusedTests in this PR, but made this method more generic. ResourceFocusedTest does this slightly differently since it compresses the repeated element and writes to the OutputStream instead of having the entire uncompressed body up front.